data_process/concatenate_dataset.py

```diff
@@ -65,16 +66,17 @@ def concatenate(input_output_dir: str, output_file: str, metadata: dict,
     """
 
     # ensure that files in file_names_to_concatenate are contained in input_output_dir:
-    for fname in file_names_to_concatenate:
-        ifname = os.path.join(input_output_dir, fname)
-        if not os.path.isfile(ifname):
-            raise FileNotFoundError(f"File {ifname} could not be found. Make sure that file {fname} is contained in the input_output_dir {input_output_dir}")
+    for input_dir in input_dirs:
+        for fname in file_names_to_concatenate:
+            ifname = os.path.join(input_dir, fname)
+            if not os.path.isfile(ifname):
+                raise FileNotFoundError(f"File {ifname} could not be found in {input_dir}.")
 
     # scan all files:
     entries_per_year = []
     entries_per_year_red = []
     for idx, fname in enumerate(file_names_to_concatenate):
-        ifname = os.path.join(input_output_dir, fname)
+        ifname = os.path.join(input_dirs[0], fname)
```

Comment:
here we check only the first directory. But I guess we assumed beforehand that the same filenames exist in each directory?